src/bin/zmtrainsa

```diff
@@ -117,6 +117,13 @@ fi
 source `dirname $0`/zmshutil || exit 1
 zmsetvars
 
+export TMPDIR=${zmtrainsa_tmp_directory:-${zimbra_tmp_directory}/zmtrainsa}
+if [ ! -d "$TMPDIR" ]; then
+    if ! mkdir "$TMPDIR"; then
```

Comment:
If $TMPDIR is empty/unset, or if directory does not exist, I think it's best that we just exit with an error.  If the directory is the standard directory ($zimbra_tmp_directory) permissions should be OK.  If it is a custom directory, then the person that customized the setting should pre-create the directory with the proper permissions for their use case.  In no case would we want to accidentally re-create zimbra_tmp_directory with bad permissions, which as a corner case, would be possible with this logic.